webapp/components/player/player.tsx

```diff
@@ -39,13 +47,108 @@ export default function Player(props: { stream: MediaStream, muted: boolean, aud
   const [currentDeviceSpeaker] = useAtom(deviceSpeakerAtom)
   const [speakerStatus] = useAtom(speakerStatusAtom)
   const refPlayPromise = useRef<Promise<void> | null>(null)
+  const refControls = useRef<HTMLDivElement>(null)
+  const [showControls, setShowControls] = useState(false)
+  const refTimeoutId = useRef<NodeJS.Timeout | null>(null)
+  const [isMuted, setIsMuted] = useState(false)
+  const [isFullscreened, setIsFullscreened] = useState(false)
+  const [isPictureInPictured, setIsPictureInPictured] = useState(false)
+
+  const handleMouseMove = () => {
+    setShowControls(true)
+    if (refTimeoutId.current) clearTimeout(refTimeoutId.current)
+    const newTimeout = setTimeout(() => {
+      setShowControls(false)
+    }, 2000)
+    refTimeoutId.current = newTimeout
+  }
+
+  const toggleMute = () => isMuted ? setIsMuted(false) : setIsMuted(true)
+
+  const isFullscreenSupported = () => {
+    const container = refVideo.current?.parentElement
+    return (
+      typeof document.exitFullscreen === 'function' &&
+    typeof container?.requestFullscreen === 'function'
+    )
+  }
```

Comment:
This can be extracted as a `const` value or a util function and placed outside of the function component, since it's the bowser's capability. Like:

```ts
export const isFullscreenSupported =  typeof document.exitFullscreen === 'function' &&
    typeof document.body.requestFullscreen === 'function'
```